security consideration for signature/mac (#245)
oauth-wg/draft-ietf-oauth-status-list: draft-ietf-oauth-status-list.md

A Status List Token in the CWT format should follow the security considerations of {{RFC8392}}.

## MAC based Deployments

> This specification allows for both, cryptographic signatures and MAC-based variants to be used to secure a Status List Token. Implementers are RECOMMENDED to match the mechanism used for the Status List Token with the one used for its Referenced Tokens. For (H)MAC based deployments, the security implications of a shared secret should be understood and should match the general security considerations of the deployment.

Reviewer: ```suggestion
This specification allows for both cryptographic signatures and message authentication codes (MAC) to secure a Status List Token. Implementers are RECOMMENDED to match the mechanism used for the Status List Token with the one used for its Referenced Tokens. For (H)MAC based deployments, the security implications of a shared secret should be understood and should match the general security considerations of the deployment.
```

> This specification allows for both cryptographic signatures and message authentication codes (MAC) to secure a Status List Token. Implementers are RECOMMENDED to match the mechanism used for the Status List Token with the one used for its Referenced Tokens. For (H)MAC based deployments, the security implications of a shared secret should be understood and should match the general security considerations of the deployment.

Reviewer: I feel like we don't underline the challenges and typical cirumstances of MAC-based setup enough yet: Issuers and Verifiers likely have a trusted relationship and a possiblity to exchange symmetric keys out-of-band

A Status List Token in the CWT format should follow the security considerations of {{RFC8392}}.

> ## MAC based Deployments

Reviewer: ```suggestion
## MAC-secured Status Lists
```

> This specification allows for both cryptographic signatures and message authentication codes (MAC) to secure a Status List Token. For (H)MAC based deployments, the security implications of a shared secret should be understood and should match the general security considerations of the deployment.

Reviewer: ```suggestion
This specification allows for both cryptographic signatures and message authentication codes (MAC) to secure a Status List Token. For (H)MAC based deployments, the security implications of a shared secret should be understood and should match the general security considerations of the deployment.```

> This specification allows for both cryptographic signatures and message authentication codes (MAC) to secure a Status List Token. Implementers are RECOMMENDED to match the mechanism used for the Status List Token with the one used for its Referenced Tokens. For (H)MAC based deployments, the security implications of a shared secret should be understood and should match the general security considerations of the deployment.

Reviewer: ```suggestion
This specification allows for both cryptographic signatures and message authentication codes (MAC) to secure a Status List Token. For (H)MAC based deployments, the security implications of a shared secret should be understood and should match the general security considerations of the deployment.
```

> This specification allows for both cryptographic signatures and message authentication codes (MAC) to secure a Status List Token. For (H)MAC based deployments, the security implications of a shared secret should be understood and should match the general security considerations of the deployment.
> There usually exists a stronger relationship between RP and Issuer for such deployments and they have the capability to securely distribute a shared secret ouf of band.

Reviewer: ```suggestion
This specification allows for both cryptographic signatures and message authentication codes (MAC) to secure a Status List Token. For (H)MAC based deployments, the security implications of a shared secret should be understood and should match the general security considerations of the deployment. There usually exists a stronger relationship between RP and Issuer for such deployments and they have the capability to securely distribute a shared secret ouf of band.
```
